### ISSUE TITLE
Fixed footer on every ioslide

### ISSUE THREAD

**Issue by martinschmelzer** [rstudio/rmarkdown#435]
Hi there,

it would be awesome to have an option to add a fixed footer on every ioslide (Like: "Author - Title of Presentation"). Fixed in the sense that if the slide has the option {.built} it should add this footer anyways and exclude it from the building process.
For adding logo at the top right corner of every slide I found a way editing the default.css. But the footer still bothers me.

kind regards,
martin

**Comment by wannesrosiers** [comment]
Could you share your solution for adding the logo at the top right corner?  I am figuring out how to change the footer and logo as well.  Thanks!

**Comment by pozelei** [comment]
Hi there,

just find de default.css inside of the ioslides folder. Then look for the following selector:

slides > slide {
  display: none;
  font-family: 'Open Sans', Arial, sans-serif;
  font-size: 26px;
  color: #797979;
  width: 900px;
  height: 700px;
  margin-left: -450px;
  margin-top: -350px;
  padding: 40px 60px;
  -webkit-border-radius: 5px;
  -moz-border-radius: 5px;
  -ms-border-radius: 5px;
  -o-border-radius: 5px;
  border-radius: 5px;
  -webkit-transition: all 0.6s ease-in-out;
  -moz-transition: all 0.6s ease-in-out;
  -o-transition: all 0.6s ease-in-out;
  transition: all 0.6s ease-in-out;
  
  background-image: url(Images/YOURLOGOFILE.png) !important;
  background-size: 200px !important;
  background-repeat: no-repeat !important;
  background-position: 93.1% 4% !important;
}

As you can see, I added some lines at the end. That's you you have to do :)

**Comment by wannesrosiers** [comment]
Thanks!  I was already able to solve it the hard and dirty way, but this is much better.

**Comment by pozelei** [comment]
You're welcome! :)

**Comment by Nicolabo** [comment]
Hi,
I also your css code and works fine.

Unfortunately, when I want to create document with `shiny widget` the .png logo disappear:

`---
title: ""
subtitle: ""  
date: "25 September 2015"
output: 
  ioslides_presentation:
    css: temp.css
    widescreen: true
runtime: shiny

---`

**Comment by samuel-rosa** [comment]
@martinschmelzer , have you found a solution to add a fixed footer?

**Comment by martinschmelzer** [comment]
@sanuek-rosa , the only thing I got is this:

```
<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.2/jquery.min.js"></script>

<script>
  $(document).ready(function() {
    $('slide:not(.title-slide)').append('<footer></footer>');
    $('footer').attr('label', 'My Amazing Footer');
    
  })
</script>

<style>
  footer:after {
    font-size: 12pt;
    content: attr(label);
    position: absolute;
    bottom: 20px;
    left: 60px;
    line-height: 1.9;
    display: block;
  }
</style>
```

This will add a line of text on the bottom left.

**Comment by martinschmelzer** [comment]
@samuel-rosa Changed the code so the footer will not be displayed on the title-slide and chapter slides :)

**Comment by samuel-rosa** [comment]
@martinschmelzer , thank you for that. I have changed the code so that the footer text is centred:

```
<style>
  footer:after {
    font-size: 12pt;
    content: attr(label);
    position: absolute;
    margin-left: auto;
    margin-right: auto;
    left: 0;
    right: 0;
    text-align:center;
    bottom: 20px;
    line-height: 1.9;
    display: block;
  }
</style>
```

**Comment by sammcq** [comment]
Having trouble getting this to work. Any additional instructions for where this code block goes?

**Comment by martinschmelzer** [comment]
You have to put the JS block and the CSS block just before your first slide.

**Comment by sammcq** [comment]
Thanks! That worked. Any idea if ioslide is still being developed?

**Comment by yihui** [comment]
@sammcq The upstream ioslides development seems to have stopped in 2013: https://code.google.com/archive/p/io-2012-slides/source/default/commits For the development here in the rmarkdown repo, I'd say it is still going on but quite sparse. This feature is probably not going to be supported officially in the `ioslides_presentation()` format.

If you are interested in other presentation frameworks, footers might be a little easier, e.g. https://github.com/yihui/xaringan/issues/34
